library/core/src/option.rs

```diff
@@ -367,43 +381,52 @@
 //! assert_eq!(res, Some(vec![2, 4, 8]));
 //! ```
 //!
-//! [`Option`] also implements the [`Product`] and [`Sum`] traits, allowing
-//! an iterator over [`Option`] values to provide the
-//! [`product`][m.product] and [`sum`][m.sum] methods.
+//! [`Option`] also implements the [`Product`][impl-Product] and
+//! [`Sum`][impl-Sum] traits, allowing an iterator over [`Option`] values
+//! to provide the [`product`][Iterator::product] and
+//! [`sum`][Iterator::sum] methods.
 //!
-//! [`Product`]: Option#impl-Product%3COption%3CU%3E%3E
-//! [`Sum`]: Option#impl-Sum%3COption%3CU%3E%3E
-//! [m.product]: crate::iter::Iterator::product
-//! [m.sum]: crate::iter::Iterator::sum
+//! [impl-Product]: Option#impl-Product%3COption%3CU%3E%3E
+//! [impl-Sum]: Option#impl-Sum%3COption%3CU%3E%3E
+//!
+//! ```
+//! let v = vec![None, Some(1), Some(2), Some(3)];
+//! let res: Option<i32> = v.into_iter().sum();
+//! assert_eq!(res, None);
+//! let v = vec![Some(1), Some(2), Some(21)];
+//! let res: Option<i32> = v.into_iter().product();
+//! assert_eq!(res, Some(42));
+//! ```
 //!
 //! ## Modifying an [`Option`] in-place
 //!
-//! These methods return a mutable reference to the contained value of a
-//! [`Some`].
+//! These methods return a mutable reference to the contained value of an
+//! [`Option<T>`]:
 //!
 //! * [`insert`] inserts a value, dropping any old contents
 //! * [`get_or_insert`] gets the current value, inserting a provided
 //!   default value if it is [`None`]
 //! * [`get_or_insert_default`] gets the current value, inserting the
-//!   default value of type `T` if it is [`None`]
+//!   default value of type `T` (which must implement [`Default`]) if it is
+//!   [`None`]
 //! * [`get_or_insert_with`] gets the current value, inserting a default
 //!   computed by the provided function if it is [`None`]
 //!
-//! [`insert`]: Option::insert
 //! [`get_or_insert`]: Option::get_or_insert
 //! [`get_or_insert_default`]: Option::get_or_insert_default
 //! [`get_or_insert_with`]: Option::get_or_insert_with
+//! [`insert`]: Option::insert
 //!
-//! These methods transfer ownership of the [`Option`].
+//! These methods transfer ownership of the contained of an [`Option`]:
```

Comment:
>  //! These methods transfer ownership of the contained of an [`Option`]:

This should be "contained value of an".

Reply:
Thanks! I think it's already fixed in the latest version. https://github.com/rust-lang/rust/pull/86211/commits/2b4a6aa1494d80564662271900fa3c73d713411e

Reply:
Ah, thank you!